app/models/resource_description_step.rb

```diff
@@ -19,8 +19,13 @@ def initialize(step = 'resource-description', title = "Resource description", su
 
   def execute context
     mediaobject = context[:mediaobject]
-    mediaobject.permalink = context[:media_object].delete(:permalink)
-    mediaobject.update_datastream(:descMetadata, context[:media_object])
+    populate_from_catalog = context[:media_object][:import_bib_record].present?
+    if populate_from_catalog
+      mediaobject.descMetadata.populate_from_catalog!(context[:media_object][:bibliographic_id])
+    else
+      mediaobject.permalink = context[:media_object].delete(:permalink)
```

Comment:
The permalink isn't stored in the MODS; it's stored elsewhere. It's also specific to the Avalon version of the resource, which doesn't exist yet at import time, so the catalog wouldn't have a reason to know about it.